### DISCUSSION TITLE
pip install broken for Debian Bookworm / Oracle MySQL

### DISCUSSION THREAD

**Discussion by lifeofguenter** [PyMySQL/mysqlclient#638] [Build issue] [answered]
### Describe the bug

When trying to install on Debian Bookworm (12) with Oracle MySQL I get the following error:

```
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [25 lines of output]
      Trying pkg-config --exists mysqlclient
      Command 'pkg-config --exists mysqlclient' returned non-zero exit status 1.
      Trying pkg-config --exists mariadb
      Command 'pkg-config --exists mariadb' returned non-zero exit status 1.
      Traceback (most recent call last):
        File "/home/guenter/Projects/panel/.venv/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/home/guenter/Projects/panel/.venv/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/home/guenter/Projects/panel/.venv/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-p5av0v8f/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 341, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-p5av0v8f/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 323, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-p5av0v8f/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 338, in run_setup
          exec(code, locals())
        File "<string>", line 154, in <module>
        File "<string>", line 48, in get_config_posix
        File "<string>", line 27, in find_package_name
      Exception: Can not find valid pkg-config name.
      Specify MYSQLCLIENT_CFLAGS and MYSQLCLIENT_LDFLAGS env vars manually
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.
```

Everything works fine with `2.1.1` so I am neither missing `pkg-config` nor am I missing any `libmysqlclient-dev` packages.

I saw in another bug report I need to fiddle around with FLAGS? But why is this suddenly necessary?

### Minimum but complete code to reproduce

```python
# Write Python code here.
import MySQLdb

conn = MySQLdb.connect(host='127.0.0.1', port=3306, user='root')
...
```

### Schema and initial data required to reproduce.

```sql
-- Write SQL here.
-- e.g. CREATE TABLE ...
```

**Comment by methane** [comment]
> Everything works fine with `2.1.1` so I am neither missing `pkg-config` nor am I missing any `libmysqlclient-dev` packages.

pkg-config is needed from 2.2. So it doesn't mean you are not missing pkg-config.



> I saw in another bug report I need to fiddle around with FLAGS? But why is this suddenly necessary?

Because MySQL broke mysql_config. I need to switch to pkg-config.

**Comment by lifeofguenter** [comment, marked as answer]
The issue (as pointed in another thread):

```
❯ pkg-config --cflags --libs mysqlclient
Package openssl was not found in the pkg-config search path.
Perhaps you should add the directory containing `openssl.pc'
to the PKG_CONFIG_PATH environment variable
Package 'openssl', required by 'mysqlclient', not found
```

not sure why openssl is required now and not before, but installing `libssl-dev` fixed the issue.

Maybe it makes sense to be more verbose with the installation so this can be caught by a user?

**Reply by methane** [reply]
Which thread?
I can not support all possible configuration. Such error should be fixed by users.
I am not free tech support.

**Reply by lifeofguenter** [reply]
??? 

The error message I posted initially does not remotely indicate any package missing. So it might be helpful to just have somewhere the output of "pkg-config --cflags --libs xxx", so a user can debug the issue easier.

Especially given that previous behavior was different.

Not sure what that has to do with free tech support at all.

**Reply by methane** [reply]
> The error message I posted initially does not remotely indicate any package missing.

`pkg-config --exists` error indicates missing pc file in most case. Your case is very rare so I assume you are missing pc file.

> Especially given that previous behavior was different.

I don't get what you mean.

> Not sure what that has to do with free tech support at all.

Fixing any broken environment in user site.
Make sure `pkg-config` working is user's responsibility and I don't want to help it.

**Comment by agghpm21** [comment]
On RHEL9 it seems that there's a different package name in pkg-config:

```
[root@qa conf]# pkg-config --exists mariadb
[root@qa conf]# echo $?
1
[root@qa conf]# pkg-config --exists mysqlclient
[root@qa conf]# echo $?
1
[root@qa conf]# pkg-config --exists libmariadb
[root@qa conf]# echo $?
0
[root@qa conf]# pkg-config --cflags --libs libmariadb
-I/usr/include/mysql/ -L/usr/lib64/ -lmariadb 

```

**Reply by agghpm21** [reply]
```
export MYSQLCLIENT_CFLAGS=`pkg-config libmariadb --cflags`
export MYSQLCLIENT_LDFLAGS=`pkg-config libmariadb --libs`
```
does the job on RHEL9/Oracle9
